go/obscuronode/enclave/enclave.go

```diff
@@ -176,7 +192,10 @@ func (e *enclaveImpl) SubmitRollup(rollup common2.ExtRollup) {
 		Header:       rollup.Header,
 		Transactions: decryptTransactions(rollup.Txs),
 	}
-	e.db.StoreRollup(&r)
+	// only store if the parent exists
+	if e.db.ExistRollup(r.Header.ParentHash) {
```

Comment:
Should we log when it's trying to submit a rollup that the parent doesn't exist ?

Reply:
I'll add something. Might help with invesitaging.
This scenario is basically when a node receives a rollup during the gossip period, and it's most likely behind.